packages/build-tools/src/utils/expoUpdatesCli.ts

```diff
@@ -22,8 +22,11 @@ export async function expoUpdatesCommandAsync(projectDir: string, args: string[]
   try {
     return (await spawnAsync(expoUpdatesCli, args)).stdout;
   } catch (e: any) {
-    if (e.stderr) {
-      if ((e.stderr as string).includes('Invalid command')) {
+    if (e.stderr && typeof e.stderr === 'string') {
+      if (
+        e.stderr.includes('Invalid command') || // SDK 51+
+        e.stderr.includes('commands[command] is not a function') // SDK <= 50
+      ) {
```

Comment:
Is it possible to determine the expo SDK we are dealing with? If we support many many old SDK's, perhaps something like `resolveFrom('expo/package.json')` and getting the version from there would be more robust

Reply:
We'd need to resolve the expo-updates package version (which we do already in some places) but I'd like to move away from if possible since things like canary versions don't work very well with it (background on that decision: https://github.com/expo/expo/pull/27143).

For that reason I prefer just trying/catching, but I could be swayed the other way.